[TASK] Deprecate INCLUDE_TYPOSCRIPT syntax (#4827)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/Configuration/TypoScript/Syntax/FileImports/Index.rst

File imports
============

..  deprecated:: 13.4
    The old school :typoscript:`<INCLUDE_TYPOSCRIPT:` syntax has been deprecated

>     and will be removed with TYPO3 14.0. See :ref:`typoscript-syntax-includes-migration`.

Reviewer: ```suggestion
    and will be removed with TYPO3 v14.0. See :ref:`typoscript-syntax-includes-migration`.
```

To structure and reuse single TypoScript snippets and not stuffing everything
into one file or record, the syntax allows loading TypoScript content from sub files.

The keyword :typoscript:`@import` is a syntax construct and
thus available in both frontend TypoScript and backend TSconfig.

> :typoscript:`@import` allows including additional files using wildcards the file

Reviewer: ```suggestion
:typoscript:`@import` allows including additional files using wildcards on the file
```

    :typoscript:`@import` statement.
    See :ref:`typoscript-syntax-includes-migration-recursive`.

..  important::
    :typoscript:`@import` supports the file endings `.typoscript` and `.tsconfig`
    only. All other file names will be ignored. Rename all TypoScript files to
    these endings.

Most usages of :typoscript:`<INCLUDE_TYPOSCRIPT:` can be turned into :typoscript:`@import`
easily. A few examples:

..  code-block:: diff

    -<INCLUDE_TYPOSCRIPT: source="FILE:EXT:my_extension/Configuration/TypoScript/myMenu.typoscript">
    +@import 'EXT:my_extension/Configuration/TypoScript/myMenu.typoscript'

     # Including .typoscript files in a single (non recursive!) directory
    -<INCLUDE_TYPOSCRIPT: source="DIR:EXT:my_extension/Configuration/TypoScript/" extensions="typoscript">
    +@import 'EXT:my_extension/Configuration/TypoScript/*.typoscript'

    # Including .typoscript and .ts files in a single (non recursive!) directory
    -<INCLUDE_TYPOSCRIPT: source="DIR:EXT:my_extension/Configuration/TypoScript/" extensions="typoscript,ts">
    +@import 'EXT:my_extension/Configuration/TypoScript/*.typoscript'

>      # Rename all files ending on .ts to .typoscript

Reviewer: ```suggestion
     # Rename all files ending from .ts to .typoscript
```

     # Including a file conditionally
    -<INCLUDE_TYPOSCRIPT: source="FILE:EXT:my_extension/Configuration/TypoScript/user.typoscript" condition="[frontend.user.isLoggedIn]">
    +[frontend.user.isLoggedIn]
    +    @import 'EXT:my_extension/Configuration/TypoScript/user.typoscript'
    +[END]

There are a few more use cases that cannot be transitioned so easily since
:typoscript:`@import` is a bit more restrictive.

As one restriction :typoscript:`@import` cannot include files from arbitrary directories
like :file:`fileadmin/`, but only from extensions by using the :typoscript:`EXT:`
prefix. Instances that use :typoscript:`<INCLUDE_TYPOSCRIPT:` with :typoscript:`source="FILE:./someDirectory/..."`

> should move this TypoScript into a project or site extension. Such instances are also encouraged to

Reviewer: ```suggestion
should move these TypoScript files into a project or site extension. Such instances are also encouraged to
```

> look into the TYPO3 v13 "Site sets" feature and eventually transition towards it along the way.

Reviewer: I suggest to link "site sets" for easy access of the reader.

Author: done

> look into the TYPO3 v13 "Site sets" feature and eventually transition towards it along the way.

Author: ```suggestion
look into the TYPO3 v13 :ref:`Site sets <site-sets>` feature and eventually transition towards it along the way.
```

> :typoscript:`@import` allows to import files with the file ending `.typoscript`

Reviewer: You also mentioned that above in an important section.

Author: Repetition cannot harm
